Libraries/ReactNative/requireNativeComponent.windows.js

```diff
@@ -0,0 +1,239 @@
+/**
+ * Copyright (c) 2015-present, Facebook, Inc.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ *
+ * @providesModule requireNativeComponent
+ * @flow
+ * @format
+ */
+'use strict';
+
+const Platform = require('Platform');
+const ReactNativeBridgeEventPlugin = require('ReactNativeBridgeEventPlugin');
+const ReactNativeStyleAttributes = require('ReactNativeStyleAttributes');
+const UIManager = require('UIManager');
+
+const createReactNativeComponentClass = require('createReactNativeComponentClass');
+const insetsDiffer = require('insetsDiffer');
+const matricesDiffer = require('matricesDiffer');
+const pointsDiffer = require('pointsDiffer');
+const processColor = require('processColor');
+const resolveAssetSource = require('resolveAssetSource');
+const sizesDiffer = require('sizesDiffer');
+const verifyPropTypes = require('verifyPropTypes');
+/* $FlowFixMe(>=0.54.0 site=react_native_oss) This comment suppresses an error
+ * found when Flow v0.54 was deployed. To see the error delete this comment and
+ * run Flow. */
+const invariant = require('fbjs/lib/invariant');
+const warning = require('fbjs/lib/warning');
+
+/**
+ * Used to create React components that directly wrap native component
+ * implementations.  Config information is extracted from data exported from the
+ * UIManager module.  You should also wrap the native component in a
+ * hand-written component with full propTypes definitions and other
+ * documentation - pass the hand-written component in as `componentInterface` to
+ * verify all the native props are documented via `propTypes`.
+ *
+ * If some native props shouldn't be exposed in the wrapper interface, you can
+ * pass null for `componentInterface` and call `verifyPropTypes` directly
+ * with `nativePropsToIgnore`;
+ *
+ * Common types are lined up with the appropriate prop differs with
+ * `TypeToDifferMap`.  Non-scalar types not in the map default to `deepDiffer`.
+ */
+import type {ComponentInterface} from 'verifyPropTypes';
+
+let hasAttachedDefaultEventTypes: boolean = false;
+
+function requireNativeComponent(
+  viewName: string,
+  componentInterface?: ?ComponentInterface,
+  extraConfig?: ?{nativeOnly?: Object},
+): React$ComponentType<any> | string {
+  function attachDefaultEventTypes(viewConfig: any) {
+    if (UIManager.ViewManagerNames) {
+      // Lazy view managers enabled.
```

Comment:
Same comment here as in UIManager.windows.js, + a nit about moving the "Platform.OS === 'android'" to the "else if". I think it changes the meaning a little, event though it may not matter in Windows.
I see you didn't remove support for exporting all constants, so perhaps the more correct change would've been to convert "if (Platform.OS === 'android')" into "if (Platform.OS === 'android' | |Platform.OS === 'windows' )"

Reply:
Same reply - wasn't too concerned w/ documentation because the same patch is being applied to react-native

---
In reply to: [174842190](https://github.com/Microsoft/react-native-windows/pull/1717#discussion_r174842190) [](ancestors = 174842190)

Reply:
Oh, did you send this to RN as well? In that case you should address the meaning change somehow.. Perhaps by converting ===android to !== ios :)

---
In reply to: [174882348](https://github.com/Microsoft/react-native-windows/pull/1717#discussion_r174882348) [](ancestors = 174882348,174842190)

Reply:
https://github.com/facebook/react-native/pull/18381

Reply:
https://github.com/facebook/react-native/pull/18381